rust/lance-index/src/vector/hnsw/builder.rs

```diff
@@ -405,9 +406,7 @@ impl HnswBuilder {
             let cur_level = HnswLevelView::new(level, nodes);
             ep = greedy_search(&cur_level, ep, &dist_calc);
         }
-
-        let mut pruned_neighbors_per_level: Vec<Vec<_>> =
-            vec![Vec::new(); (target_level + 1) as usize];
+        let mut rev_edges: Vec<(u32, u16, OrderedFloat, u32)> = Vec::new();
```

Comment:
Can we have a struct for this tuple?

Reply:
Yes we can. Done.

The default value for chunk size should be a constant, and this is an oversight on the part of the ParlayANN authors IMO. The way I've done it, it should be bounded by `ef_construction` (or some multiple of it) so that the quadratic complexity of computing the "additional candidate edges" does not explode beyond what is already being done as part of the searches (which, very loosely speaking, each take time more or less proportional to `ef_construction`).
Given that `ef_construction` should typically be set at a few hundred for any reasonably sized dataset, I don't think this will cause any problems with limiting parallelism. I added a warning though.